File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/book/updateBookRequest.ts
import { Book, CoverType } from "../../../..";

export interface UpdateBookRequest {
    id: number;
    name: string;
    publicationDate: Date;
    price: number;
    coverType: CoverType;
    pageAmount: number;
    coverImgUrl: string,
    description: string | null,
    authorId: number;
    genreId: number;
    publisherId: number;
}
export function mapBookToUpdateBookRequest(book: Book): UpdateBookRequest {
    return {
        id: book.id,
        name: book.name,
        publicationDate: book.publicationDate,
        price: book.price,
        coverType: book.coverType,
        pageAmount: book.pageAmount,
        coverImgUrl: book.coverImgUrl,
        description: book.description,
        authorId: book.author.id,
        genreId: book.genre.id,
        publisherId: book.publisher.id
    }
}

File: src/ELibrary.Backend/LibraryApi/Domain/Dtos/Book/UpdateBookRequest.cs
using LibraryShopEntities.Domain.Entities.Library;

namespace LibraryApi.Domain.Dto.Book
{
    public class UpdateBookRequest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime PublicationDate { get; set; }
        public decimal Price { get; set; }
        public CoverType CoverType { get; set; }
        public string CoverImgUrl { get; set; }
        public int PageAmount { get; set; }
        public int StockAmount { get; set; }
        public int AuthorId { get; set; }
        public int GenreId { get; set; }
        public int PublisherId { get; set; }
    }
}
